Jellyfin.Api/Helpers/RequestHelpers.cs

```diff
@@ -55,6 +56,32 @@ public static (string, SortOrder)[] GetOrderBy(IReadOnlyList<string> sortBy, IRe
             return result;
         }
 
+        /// <summary>
+        /// Checks if the user can access a user.
+        /// </summary>
+        /// <param name="claimsPrincipal">The <see cref="ClaimsPrincipal"/> for the current request.</param>
+        /// <param name="userId">The user id.</param>
+        /// <returns>A <see cref="bool"/> whether the user can update the entry.</returns>
+        internal static Guid GetUserId(ClaimsPrincipal claimsPrincipal, Guid? userId)
+        {
+            var authenticatedUserId = claimsPrincipal.GetUserId();
+
+            // UserId not provided, fall back to authenticated user id.
+            if (userId is null || userId.Value.Equals(default))
+            {
+                return authenticatedUserId;
+            }
+
+            // User must be administrator to access another user.
+            var isAdministrator = claimsPrincipal.IsInRole(UserRoles.Administrator);
+            if (!userId.Value.Equals(authenticatedUserId) && !isAdministrator)
+            {
+                throw new SecurityException("Forbidden");
+            }
+
+            return userId.Value;
```

Comment:
Should we check if the provided user id is valid before returning it here? I believe right now the API throws a 500 with an incorrect user id.

Reply:
Actually at that point we might even change it to `GetUser()` and return the User(Dto?) object.

Reply:
I think that change is out of scope for this PR as it looks like there are a good amount of endpoints that still function if the user doesn't exist